clash-lib/src/Clash/Backend/VHDL.hs

```diff
@@ -244,18 +244,26 @@ instance Backend VHDLState where
     let attrs = [ (id_, attr)
                 | NetDecl' _ _ id_ (Right hwtype) _ <- ds
                 , attr <- hwTypeAttrs hwtype]
+        cds = mapMaybe mCompDecl ds
     if isEmpty decs
        then insts ds
        else nest 2
               (pretty nm <+> colon <+> "block" <> line <>
                pure decs <>
-               if null attrs
+               (if null attrs
                 then emptyDoc
-                else line <> line <> renderAttrs (TextS.pack "signal") attrs) <> line <>
+                else line <> line <> renderAttrs (TextS.pack "signal") attrs) <>
+               if null cds
+                then emptyDoc
+                else line <> line <> renderCompDecls (mapMaybe mCompDecl ds)) <>
```

Comment:
I would still do this with `inst_`, i.e. `filter isCompDecl` (instead of the `mapMaybe`) then pass the whole declaration to `inst_` instead of doing this extra unpacking to a tuple